sdk-internals/communicator/ssh/communicator.go

```diff
@@ -1024,3 +1035,24 @@ func scpUploadDir(root string, fs []os.FileInfo, w io.Writer, r *bufio.Reader) e
 
 	return nil
 }
+
+// isExcluded checks if the given file path is excluded by the given list of
+// excludes.
+// It does shell style matching, so the excludes can contain wildcards.
+func isExcluded(filePath string, excludes []string) bool {
```

Comment:
I need to add the change to WinRM comm as well.

Reply:
In addition to winrm, I would also suggest modifying the `sftp` methods for uploading a directory in order to take this into account.

I spent a bit of time in the communicator package recently for SCP/SFTP feature parity, as such I may be able to help if needed, please let me know if you want help on this!

Reply:
Also added this check for [sftp upload](https://github.com/Glyphack/packer-plugin-sdk/blob/d59871b6926dab92ed60f060d577c20b655461af/sdk-internals/communicator/ssh/communicator.go#L543)